plugins/conditionalrestrictions.py

```diff
@@ -0,0 +1,210 @@
+##                                                                       ##
+## This program is distributed in the hope that it will be useful,       ##
+## but WITHOUT ANY WARRANTY; without even the implied warranty of        ##
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         ##
+## GNU General Public License for more details.                          ##
+##                                                                       ##
+## You should have received a copy of the GNU General Public License     ##
+## along with this program.  If not, see <http://www.gnu.org/licenses/>. ##
+##                                                                       ##
+###########################################################################
+
+from modules.OsmoseTranslation import T_
+from plugins.Plugin import Plugin
+import re
+from datetime import date
+from modules.Stablehash import stablehash64
+
+class ConditionalRestrictions(Plugin):
+  def init(self, logger):
+    Plugin.init(self, logger)
+
+    self.ReYear = re.compile(r'20\d\d') # Update in 2099
+    self.currentYear = date.today().year
+
+    self.errors[33501] = self.def_class(item = 3350, level = 2, tags = ['highway', 'fix:chair'],
+        title = T_('Bad conditional restriction'),
+        detail = T_('''Conditional restrictions should follow `value @ condition; value2 @ condition2` syntax.
+Combined restrictions should follow `value @ (condition1 AND condition2)`.
+Parentheses `()` must be used around the condition if the condition itself contains semicolons `;`, i.e. `value @ (date;date)`.'''))
+    self.errors[33502] = self.def_class(item = 3350, level = 3, tags = ['highway', 'fix:chair'],
+        title = T_('Combine conditions using `AND`'),
+        detail = T_('''`AND` (uppercase) is to be preferred over lowercase variants when combining restrictions.'''))
+    self.errors[33503] = self.def_class(item = 3350, level = 3, tags = ['highway', 'fix:chair'],
+        title = T_('Expired conditional'),
+        detail = T_('''This conditional was only valid up to a date in the past. It can likely be removed.'''),
+        trap = T_('''Other tags might need to be updated too to reflect the new situation.'''))
+
+  def way(self, data, tags, nds):
+    # Get the relevant tags with *:conditional
+    tags_conditional = {}
+    for tag in tags:
+      if tag[-12:] == ":conditional":
+        if "source:" in tag or "note:" in tag or "fixme:" in tag:
+          continue
+        tags_conditional[tag] = tags[tag]
+    if len(tags_conditional) == 0:
+      return
+
+    err = []
+    for tag in tags_conditional:
+      tag_value = tags_conditional[tag]
+      conditions = []
+      parentheses = 0
+      bad_tag = False
+
+      if not "@" in tag_value:
+        err.append({"class": 33501, "subclass": 0 + stablehash64(tag + '|' + tag_value), "text": T_("Missing `@` in \"{0}\"", tag)})
+        continue
+
```

Comment:
I'm wondering if we should also check that we have ` @ ` in the tag value.

Reply:
Can be done fairly easily in a crude way:
`if tag_value.count("@") != tag_value.count(" @ "):`

I'm not so sure how much that adds though; this is a thing that bots can fix safely without needs for humans (unlike 'and' which may hypothetically also occur in a date-time range comment), and it's only readability, not invalid.

Reply:
(To avoid having to make a new branch again I'll await your and/or Frodrigo's comment on this so I can put everything in one patch ;) )